src/Compilers/Core/MSBuildTask/MvidReader.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    public sealed class MvidReader : BinaryReader
+    {
+        public MvidReader(Stream stream) : base(stream)
+        {
+        }
+
+        public static Guid ReadAssemblyMvid(Stream stream)
+        {
+            try
+            {
+                var mvidReader = new MvidReader(stream);
+                return mvidReader.FindMvid();
+            }
+            catch (EndOfStreamException)
+            {
+            }
+
+            return Guid.Empty;
```

Comment:
@jaredpar I renamed the methods. Let me know if that's better.

---
In reply to: [114092229](https://github.com/dotnet/roslyn/pull/19133#discussion_r114092229) [](ancestors = 114092229)